src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/PtvServiceChannelProvider.java

```diff
@@ -235,15 +244,69 @@ public ServiceLocationServiceChannel findServiceLocationServiceChannel(ServiceLo
         waitServiceChannelUpdate(updatedPtvChannelId);
         return findPhoneChannelAfterUpdate(phoneChannelId);
       } else {        
-        logger.severe(() -> String.format(FAILED_TO_UPDATE_SERVICE_LOCATION_SERVICE_CHANNEL, response.getStatus(), response.getMessage()));
+        logger.severe(() -> String.format(FAILED_TO_UPDATE_SERVICE_CHANNEL, response.getStatus(), response.getMessage()));
         return IntegrationResponse.statusMessage(response.getStatus(), response.getMessage());
       }
       
     }
     
     return null;
   }
+  
+
+  @Override
+  public IntegrationResponse<PrintableFormServiceChannel> updatePrintableFormServiceChannel(PrintableFormServiceChannelId printableFormChannelId, PrintableFormServiceChannel printableFormServiceChannel) {
+    PrintableFormServiceChannelId ptvPrintableFormServiceChannelId = idController.translatePrintableFormServiceChannelId(printableFormChannelId, PtvConsts.IDENTIFIER_NAME);
+    if (ptvPrintableFormServiceChannelId != null) {
+      V7VmOpenApiPrintableFormChannel ptvPrintableFormServiceChannel = loadServiceChannel(ServiceChannelType.PRINTABLE_FORM, ptvPrintableFormServiceChannelId.getId());
+      if (ptvPrintableFormServiceChannel == null) {
+        return IntegrationResponse.statusMessage(Response.Status.BAD_REQUEST.getStatusCode(), COULD_NOT_RESOLVE_SERVICE_CHANNEL);
+      }
+      
+      V7VmOpenApiPrintableFormChannelInBase ptvPrintableFormServiceChannelIn = ptvOutPtvInTranslator.translatePrintableFormChannel(ptvPrintableFormServiceChannel);
+      
+      if (ptvPrintableFormServiceChannelIn == null) {
+        logger.log(Level.SEVERE, () -> String.format("Failed to translate ptv %s service location service channel", ptvPrintableFormServiceChannelId.getId()));
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Use already-defined constant 'FAILED_TO_TRANSLATE_PTV_SERVICE_LOCATION_SERVICE_CHANNEL' instead of duplicating its value here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)